### DISCUSSION TITLE
Auto retry feedback

### DISCUSSION THREAD

**Discussion by hardysabs2** [sindresorhus/got#2091] [General]
I understand a got request (though you can specify retries explicitly) implicitly includes under the hood auto retry (3 times?).  Is this right?  Is it possible to get feedback on whether retries actually occurred?  For success in the response object, for failure in the caught error object? 

Are auto retries attempted for any kind of failure, e.g. any of the timeouts or any other reason?  Is there any kind of back-off / delay algorithm employed before the retry?

Apologies I was struggling to find this in the documentation.

**Comment by szmarczak** [comment]
<details>
<summary>(3 times?)</summary>

![image](https://user-images.githubusercontent.com/36894700/180518552-59d1dadf-7adc-48c8-9f10-5a3d3d9f11a3.png)

---

![image](https://user-images.githubusercontent.com/36894700/180519396-779bd418-2f4e-4ae8-8b7b-d0c7baa4e293.png)

---

![image](https://user-images.githubusercontent.com/36894700/180519490-4916e666-a201-4ad7-bf85-90706f71e336.png)

</details>

---

<details>
<summary>Is it possible to get feedback on whether retries actually occurred?</summary>

![image](https://user-images.githubusercontent.com/36894700/180519165-ac84c669-0e4a-422f-92bd-ef4aef7e5bfa.png)

---

![image](https://user-images.githubusercontent.com/36894700/180519212-9245ee0a-21d8-4803-9cf5-231ef5ec1224.png)

</details>

---

<details>
<summary>for failure in the caught error object?</summary>

![image](https://user-images.githubusercontent.com/36894700/180519868-bc6d4f28-fb58-4e65-8ac5-b734ce742995.png)

---

![image](https://user-images.githubusercontent.com/36894700/180519948-96c4a1d3-6f7f-40c4-b34b-e7601853c528.png)

</details>

---

<details>
<summary>Are auto retries attempted for any kind of failure, e.g. any of the timeouts or any other reason?</summary>

![image](https://user-images.githubusercontent.com/36894700/180518552-59d1dadf-7adc-48c8-9f10-5a3d3d9f11a3.png)

---

![image](https://user-images.githubusercontent.com/36894700/180520513-c01d533d-ffce-4b18-987d-451365b85b84.png)

</details>

---

<details>
<summary>Is there any kind of back-off / delay algorithm employed before the retry?</summary>

![image](https://user-images.githubusercontent.com/36894700/180518552-59d1dadf-7adc-48c8-9f10-5a3d3d9f11a3.png)

---

![image](https://user-images.githubusercontent.com/36894700/180518712-8310784b-0a4a-4470-b20b-66241e2c3739.png)

---

![image](https://user-images.githubusercontent.com/36894700/180518750-2eac003f-3ec3-4583-a60f-626742b8d1d2.png)

</details>

---

> I was struggling to find this in the documentation.

Can you tell me where did you exactly look to find this information please?

**Reply by daverickdunn** [reply]
It'd be really helpful if the [retry docs](https://github.com/sindresorhus/got/blob/main/documentation/7-retry.md) mentioned this default retry behaviour. It's only mentioned in [options](https://github.com/sindresorhus/got/blob/main/documentation/tips.md) and [tips](https://github.com/sindresorhus/got/blob/main/documentation/tips.md)

**Reply by szmarczak** [reply]
IIRC we used to have one, not sure what happened

**Reply by szmarczak** [reply]
Indeed, it was in v11 but the note is missing in v12, lemme fix that
